src/NuGet.Core/NuGet.Packaging/Core/FrameworkNameValidatorUtility.cs

```diff
@@ -34,6 +34,51 @@ internal static bool IsValidFrameworkName(string path)
             return fx != null && fx.Framework != NuGetFramework.UnsupportedFramework.Framework;
         }
 
+        internal static bool IsDottedFrameworkVersion(string path)
+        {
+            foreach (string knownFolder in PackagingConstants.Folders.Known)
+            {
+                string folderPrefix = knownFolder + System.IO.Path.DirectorySeparatorChar;
```

Comment:
It's not clear to me if `path` represents filesystem path or package path. If it's nupkg path, then `Path.DIrectorySeparatorChar` is not safe, because the zip may be using a different separator.

Reply:
It's a filesystem path!